### PR TITLE
Add react-appinsights

### PR DESCRIPTION
**Pull request by pviotti**
This PR adds react-appinsights to the demo project.  
[React-appinsights](https://github.com/Azure/react-appinsights/tree/typescript-rewrite) is a small wrapper of Application Insights SDK that adds features that are used in React applications, such as automated page tracking through react-router-dom history and React components usage statistics.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -5,7 +5,9 @@
   "dependencies": {
     "@microsoft/applicationinsights-web": "^1.0.0-beta.11",
     "react": "^16.6.3",
+    "react-appinsights": "^3.0.0-beta.3",
```
[the remark this answers is not in the document]
Comment (pviotti) [reply package.json]: yep (it has a beta tag in the npm repository): `npm install --save react-appinsights@beta` or `yarn add react-appinsights@beta`

File: src/App.jsx [hunk 2]
```diff
@@ -1,18 +1,21 @@
...
+import { ReactAI, withAITracking } from "react-appinsights";
+import { Router } from 'react-router-dom';
+import { createBrowserHistory } from 'history';
 import './App.css';
 
+const history = createBrowserHistory()
 let INSTRUMENTATION_KEY = '4a795cb3-5b9e-4428-8777-0441b7ae7dc8'; // Enter your instrumentation key here
 
-let appInsights = new ApplicationInsights({config: {
+ReactAI.initialize({
```
Comment (jpiyali) [inline src/App.jsx]: In terms of work done by initialize, this method is essentially initializing the Web SKU with inputs and doing the react-appinsights initialization. What this is also doing is removing the ability for customer to use any other application insights extension, which does not seem necessary. Looking at the beta code, rootInstance is returning the ai instance initialized.

Here is a list of changes that you would need to do:
1. Add telemetryinitializer to update context properties. It seems you would like to the whole context setup, I can look into adding a public property for this on the properties extension so you can continue with you code as before.
2. You already have customDimensioninitializer that is essentially taking advantage of telemetryInitializer.

Let me know if you need help updating to implement ITelemetryPlugin, happy to setup some time for this.

**Comment by markwolff** [comment]
It might be better to show react appinsights in a separate demo that has routing and can use all of the react specific features in a nice way. One of these can be used as a starting point so only adding the SDK will be needed. https://reactjs.org/community/examples.html

**Comment by pviotti** [comment]
Waiting for the completion of [this PR](https://github.com/Azure/react-appinsights/pull/39), I've prepared a version of this repo supporting react-appinsights, [here](https://pviotti.github.io/react-appinsights-demo/) ([here ](https://github.com/pviotti/react-appinsights-demo) is the source). I'll move this demo project to the Azure-Samples org.

**Comment by hectorhdzg** [comment]
Closing in favor of https://github.com/Azure-Samples/application-insights-react-demo/pull/4/files
Thanks @pviotti

**Pull request closed without merging**